indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentTransactionalReplaceAction.java

```diff
@@ -141,22 +144,31 @@ private void tryUpgradeOverlappingPendingSegments(Task task, TaskActionToolbox t
     final SupervisorManager supervisorManager = toolbox.getSupervisorManager();
```

Comment:
General comment: this task action in particular is a bit more involved to follow. I found the PR description quite useful to understand the high-level flow. I would suggest including a version of that as javadocs in this class itself.

Reply:
Done